src/cljx/schema/core.cljx

```diff
@@ -515,6 +487,83 @@
      [(if (= pred :else) (constantly true) pred) schema])
    (if (odd? (count preds-and-schemas)) (last preds-and-schemas))))
 
+(defprotocol HasPrecondition
+  (precondition [this]
+    "Return the Precondition for this schema.
+     A Precondition is a function of a value that returns a
+     ValidationError or nil (see spec.core)"))
+
+(extend-protocol HasPrecondition
+  schema.spec.leaf.LeafSpec
+  (precondition [this]
+    (:pre this))
+
+  schema.spec.variant.VariantSpec
+  (precondition [this]
```

Comment:
:goat: it seems like we're jumping through some hoops to make this a proper Precondition, which it isn't really.  Maybe it would be cleaner to just make it a simple predicate?